migration_steps/transform_casrec/transform/app/utilities/basic_data_table.py

```diff
@@ -21,6 +21,8 @@ def get_source_table(mapping_dict):
     no_dupes = list(set(source_table_list))
     if len(no_dupes) == 1:
         return list(set(source_table_list))[0]
+    elif len(no_dupes) == 0:
```

Comment:
NB my PR for IN-1061 includes the same code as you have to fall back to table definitions if there is no casrec mapping for a column: https://github.com/ministryofjustice/opg-data-casrec-migration/pull/491/files#diff-1b140eca9ba2743eac7594a54b4b8cd316bf7a28b154e2c9fa0fde43b90b7058

What do you think about putting in some of the additional checking and logging as used there to ensure we get a "good" value back from the function?

Reply:
I think `get_source_table()` is used in just one place so I thought there's no point making `table_definition` an optional argument. If the key `source_table_name` doesn't exist in `table_definition` I think the code will bomb out with an error at that point, which is fine imo as all table definitions should have that key especially if the source table isn't defined anywhere else in the mapping.